core/includes/mail.inc

```diff
@@ -186,7 +186,7 @@ function backdrop_mail($module, $key, $to, $language, $params = array(), $reply
       // Log errors.
       if (!$message['result']) {
         watchdog('mail', 'Error sending e-mail (from %from to %to).', array('%from' => $message['from'], '%to' => $message['to']), WATCHDOG_ERROR);
-        backdrop_set_message(t('Unable to send e-mail. Contact the site administrator if the problem persists.'), 'error');
```

Comment:
While the "contact an administrator" isn't helpful (for reasons mentioned in the main issue), it's at least something actionable. There are a lot of reasons why mail may fail to send: invalid credentials, misconfigured local server, misconfigured remote server, no such domain or user, blocked for spam/virus/graylisting, message too long. Some of these can be addressed by the local site admin, others require a server admin or the receiver's admin to fix. Even if we don't put it in the on-screen message, we should add something to the watchdog logs. As the admin, please give me a clue.

Reply:
I also hesitated a bit looking at that change, but if sending mail failed, there _will_ be a watchdog entry anyway with more information. To my understanding the end user (the one that tried to send a mail) won't need much more information than "it failed".

Reply:
The only watchdog message guaranteed here is "Error sending email (from %from to %to.)"  Does `$system` keep a last error message that we can include in the logs?  If not, that's something we should add to the mail handler API (and is out-of-scope for this issue.)

Reply:
> The only watchdog message guaranteed here is "Error sending email (from %from to %to.)"

Depending on what's used for sending mails, there _will_ be additional ones. I don't think, there are more details available in function backdrop_mail(), which we could log. And getting more goes far beyond the scope of this PR - which tries to deal with wording.

But feel free to open an issue for whatever you're missing when sending mail fails.

Reply:
I'd say a new issue is needed for this.